test-chart/action.yaml

```diff
@@ -42,38 +27,33 @@ runs:
       id: get-version
       shell: bash
       run: |
-        VERSION=$(curl -s https://api.github.com/repos/silphid/testchart/releases/latest | grep tag_name | cut -d '"' -f 4 | sed 's/^v//')
-        echo "Latest version of testchart is v$VERSION"
-        echo "version=$VERSION" >> $GITHUB_OUTPUT
+        if [ -z "${{ inputs.version }}" ]; then
+          VERSION=$(curl -s https://api.github.com/repos/silphid/testchart/releases/latest | grep tag_name | cut -d '"' -f 4 | sed 's/^v//')
+          echo "Latest version of testchart is v$VERSION"
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+        else
+          echo "version=${{ inputs.version }}" >> $GITHUB_OUTPUT
+        fi
 
     - name: Perform chart tests
       shell: bash
       run: |
         cd ${{ inputs.work-dir }}
-        if [ -d "${{ inputs.path }}" ]; then
-
-          # Install testchart
-          BIN_DIR=$HOME/bin
-          mkdir -p $BIN_DIR
-          VERSION=${{ steps.get-version.outputs.version }}
-          curl -sLo $BIN_DIR/testchart.tar.gz https://github.com/silphid/testchart/releases/download/v$VERSION/testchart_${VERSION}_linux_amd64.tar.gz
-          tar xvzf $BIN_DIR/testchart.tar.gz -C $BIN_DIR
-
-          # Build testchart command
-          command="$BIN_DIR/testchart run \
-            --namespace ${{ inputs.namespace }} \
-            --release ${{ inputs.release }} \
-            --path ${{ inputs.path }} \
-            --show-values ${{ inputs.show-values }}"
-
-          # Add ignore patterns passed as a multi-line string, if any
-          while IFS= read -r line; do
-            command+=" --ignore '$line'"
-          done <<< "${{ inputs.ignore }}}"
-
-          # Run testchart
-          eval "$command"
-
-        else
-          echo "No ${{ inputs.path }} sub-directory found in chart"
-        fi
+        
```

Comment:
This loses the check for existence of test directory, as tests should only be run when they are present. We could simply reintroduce the check, but with an early return, instead of full indentation.

Reply:
I find that a bit redundant honestly, if someone has added the test-chart step to their workflow, then they want to run tests! If tests are missing then the CLI should be throwing an error saying there are no tests, not the action.

Reply:
That was to keep workflows using that action generic, regardless of whether they include tests or not. It could be argued that not having tests is not an error condition (eg: `go test ./...` won't complain if there are no tests). It's really up for debate, both could make sense, but I don't mind that much! ;)